### PR TITLE
add set of authorization plugins

### PR DESCRIPTION
**Pull request by vladak**
This is a set of authorization plugins written by @tulinkry.

### CODE REVIEW

File: plugins/LdapPlugin/src/opengrok/auth/plugin/LdapFilter.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     * Special values are:
+     * <ul>
+     * <li>%uid% - to be replaced with LDAP uid value</li>
+     * <li>%mail% - to be replaced with LDAP mail value</li>
+     * <li>%username% - to be replaced with OSSO user value</li>
+     * <li>%guid% - to be replaced with OSSO guid value</li>
+     * </ul>
+     *
+     * Use \% for printing the '%̈́' character.
+     *
```
Comment (tulinkry) [inline plugins/LdapPlugin/src/opengrok/auth/plugin/LdapFilter.java]: This doc actualy lacks info that it replaces any other ldap attribute which would not be ambiguous (lines 119-126)

Comment (vladak) [reply plugins/LdapPlugin/src/opengrok/auth/plugin/LdapFilter.java]: you can add it in later PR

**Comment by vladak** [comment]
Also, the NetBeans projects are probably not working out of the box, can fix them later.

File: plugins/LdapPlugin/src/opengrok/auth/plugin/LdapFilter.java (hunk 1 above)
Comment (vladak) [reply plugins/LdapPlugin/src/opengrok/auth/plugin/LdapFilter.java]: or just send me the diff and I will slap on the top

**Comment by tulinkry** [comment]
Just to be sure, so it eventually builds the plugins into a jar file and runs the tests or?

**Comment by tulinkry** [comment]
And why is junit needed? I thought we already have it since we run some tests

**Comment by vladak** [comment]
The tests of the plugins are actually not run. I've added the junit csets as a basis for running the tests via `ant` - not directly related to the auth plugins yet.

**Comment by vladak** [comment]
One more note w.r.t. junit - on Travis the tests are run via Maven which I guess knows where to get Junit.jar from. For `ant` though, you had to download the junit.jar yourself and place it in the `lib` directory. It's sort of surprising that I got to fix this only now since it was a nuisance for me from the very start.

**Comment by tulinkry** [comment]
ah right, but doing that from netbeans downloads the dependencies automaticaly when you add the library in the test dependencies, right?

**Comment by vladak** [comment]
yep, in NB this is easily fixable.

**Comment by tulinkry** [comment]
I can't see that maybe I'm just blind but did you delete the file "plugins.jar" from testdata? It seems to me that it is not used anymore.

**Comment by vladak** [comment]
It was an attempt to fix #1859 which failed, reverted and renamed test plugins to `testplugins.jar` so they are not confused with `plugins.jar` which actually gets shipped.

**Comment by vladak** [comment]
@tulinkry mentioned on Slack that he has not further comments, merging.

**Pull request merged**